### PR TITLE
Logging-metrics with status modification & getting workspaceId from workspaceToken 

### PR DESCRIPTION
**Pull request by saurav-malani**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: regulation-worker/internal/client/client.go [hunk 1]
```diff
@@ -12,105 +12,136 @@ import (
...
 	}
 
 	req.SetBasicAuth(j.WorkspaceToken, "")
 	req.Header.Set("Content-Type", "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	pkgLogger.Debugf("making request: %w", req)
+	fmt.Println("using client to get job")
+	fmt.Println("client=", j.Client)
```
Comment (lvrach) [inline regulation-worker/internal/client/client.go:53]:
```suggestion
```

File: regulation-worker/internal/client/client.go [hunk 2]
```diff
@@ -12,105 +12,136 @@ import (
...
 func (j *JobAPI) Get(ctx context.Context) (model.Job, error) {
 
+	pkgLogger.Debugf("making http request to regulation manager to get new job")
 	ctx, cancel := context.WithTimeout(ctx, time.Duration(time.Minute))
 	defer cancel()
 
 	method := "GET"
 	genEndPoint := "/dataplane/workspaces/{workspace_id}/regulations/workerJobs"
 	url := fmt.Sprint(j.URLPrefix, prepURL(genEndPoint, j.WorkspaceID))
+	pkgLogger.Debugf("making GET request to URL: %w", url)
```
Comment (lvrach) [inline regulation-worker/internal/client/client.go:40]: %w should be used for errors

File: regulation-worker/internal/delete/api/api.go [hunk 3]
```diff
@@ -26,32 +29,43 @@ type APIManager struct {
...
 
 	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewReader(reqBody))
 	if err != nil {
+		pkgLogger.Errorf("error while create new request: %v", err)
 		return model.JobStatusFailed
 	}
 	req.Header.Set("Content-Type", "application/json")
 
-	//TODO: log error received from server
+	fileCleaningTime := stats.NewTaggedStat("file_cleaning_time", stats.TimerType, stats.Tags{"jobId": fmt.Sprintf("%d", job.ID), "workspaceId": job.WorkspaceID, "destType": "api", "destName": destName})
```
Comment (lvrach) [inline regulation-worker/internal/delete/api/api.go:53]:
```suggestion
	fileCleaningTime := stats.NewTaggedStat("file_cleaning_time", stats.TimerType, stats.Tags{
	    "jobId": fmt.Sprintf("%d", job.ID), 
	    "workspaceId": job.WorkspaceID, 
	    "destType": "api", 
	    "destName": destName
	})
```

File: regulation-worker/cmd/main_test.go [hunk 4]
```diff
@@ -42,23 +41,21 @@ func handler() http.Handler {
...
-		logger.Init()
 		backendconfig.Init()
 		c := m.Run()
 		svcCancel()
 		code <- c
 
 	}()
 	<-testDataInitialized
-	_ = os.Setenv("workspaceID", workspaceID)
+	// _ = os.Setenv("workspaceID", workspaceID)
```
Comment (lvrach) [inline regulation-worker/cmd/main_test.go:58]:
```suggestion
```

File: regulation-worker/internal/client/client.go [hunk 5]
```diff
@@ -12,105 +12,133 @@ import (
...
+		pkgLogger.Errorf("error while create new http request: %v", err)
 		return model.Job{}, err
 	}
 
 	req.SetBasicAuth(j.WorkspaceToken, "")
 	req.Header.Set("Content-Type", "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	pkgLogger.Debugf("making request: %v", req)
```
Comment (lvrach) [inline regulation-worker/internal/client/client.go:51]: We already do 	`pkgLogger.Debugf("making GET request to URL: %v", url)` one of those are redundant.

**Review by lvrach — APPROVED** [review]

File: regulation-worker/internal/destination/destination.go [hunk 6]
```diff
@@ -35,6 +58,38 @@ func (d *DestMiddleware) GetDestDetails(destID string) (model.Destination, error
...
+	var err error
+	bo := backoff.NewExponentialBackOff()
+	boCtx := backoff.WithContext(bo, ctx)
+	bo.MaxInterval = time.Minute
+	bo.MaxElapsedTime = maxWait
+	var config backendconfig.ConfigT
+	var ok bool
+	if err = backoff.Retry(func() error {
+		config, ok = d.Dest.Get()
+		pkgLogger.Debugf("trying to update status...")
```
Comment (chandumlg) [inline regulation-worker/internal/destination/destination.go:82]: Move the log up (before Get())
Correct the log statement. `Fetching backend config`.

**Review by chandumlg — APPROVED** [review]

**Pull request merged**